pkg/inject/config_override.go

```diff
@@ -0,0 +1,345 @@
+package inject
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	l5dcharts "github.com/linkerd/linkerd2/pkg/charts/linkerd2"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/linkerd/linkerd2/pkg/util"
+	log "github.com/sirupsen/logrus"
+	k8sResource "k8s.io/apimachinery/pkg/api/resource"
+)
+
+type ConfigLookup interface {
```

Comment:
I think we can also get rid of this interface by making these into freestanding functions as well which accept their inputs as function arguments.